src/dbnode/encoding/m3tsz/encoder.go

```diff
@@ -635,6 +406,323 @@ func (enc *encoder) segment(resType resultType) ts.Segment {
 	return ts.NewSegment(head, tail, ts.FinalizeHead)
 }
 
+// TimestampEncoder encapsulates the state required for a logical stream of
+// bits that represent a stream of timestamps compressed using delta-of-delta
+type TimestampEncoder struct {
+	PrevTime       time.Time
+	PrevTimeDelta  time.Duration
+	PrevAnnotation []byte
+
+	Opts encoding.Options
```

Comment:
nit: Call this `Options` when it's an exported field rather than the abbreviation?